library/core/src/num/nonzero.rs

```diff
@@ -895,3 +895,63 @@ macro_rules! nonzero_unsigned_is_power_of_two {
 }
 
 nonzero_unsigned_is_power_of_two! { NonZeroU8 NonZeroU16 NonZeroU32 NonZeroU64 NonZeroU128 NonZeroUsize }
+
+macro_rules! nonzero_constants_signed {
+    ( $( $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                #[unstable(feature = "nonzero_min_max", issue = "89065")]
+                #[doc = concat!("The maximum value for a`", stringify!($Ty), "` is the same as `", stringify!($Int), "`")]
+                /// # Examples
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MAX, ", stringify!($Int), "::MAX);")]
+                pub const MAX : $Ty = $Ty::new(<$Int>::MAX).unwrap() ;
+                #[unstable(feature = "nonzero_min_max", issue = "89065")]
+                #[doc = concat!("The minimum value for a`", stringify!($Ty), "`.")]
+                /// # Examples
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MIN, ", stringify!($Int), "::MIN;")]
+                pub const MIN : $Ty = $Ty::new(<$Int>::MIN).unwrap();
+                }
+        )+
+    }
+}
+
+nonzero_constants_signed! {
+    NonZeroI8(i8);
+    NonZeroI16(i16);
+    NonZeroI32(i32);
+    NonZeroI64(i64);
+    NonZeroI128(i128);
+    NonZeroIsize(isize);
+}
+
+macro_rules! nonzero_constants_unsigned{
+    ( $( $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                #[unstable(feature = "nonzero_min_max", issue = "89065")]
+                #[doc = concat!("The maximum value for a`", stringify!($Ty), "` is the same as `", stringify!($Int), "`")]
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MAX, ", stringify!($Int), "::MAX);")]
+                /// Note, while most integer types are defined for every whole number between MIN and
```

Comment:
I think this should be either `Note that, while` or `Note: while` rather than `Note, while`, but I'm not sure whether others would agree with that grammar nitpick.

Reply:
Yeah, I'd vote for `Note: While` with a capital `W`.